src/aspire/source/simulation.py

```diff
@@ -81,12 +82,17 @@ def __init__(
         if unique_filters is None:
             unique_filters = []
         self.unique_filters = unique_filters
+        # sim_filters must be a deep copy so that it is not changed
+        # when unique_filters is changed
+        self.sim_filters = copy.deepcopy(unique_filters)
 
         # Create filter indices and fill the metadata based on unique filters
         if unique_filters:
             if filter_indices is None:
                 filter_indices = randi(len(unique_filters), n, seed=seed) - 1
             self.filter_indices = filter_indices
+        else:
+            self.filter_indices = np.array([0] * n)
```

Comment:
Trying to understand the need for this. Could you help me out?

Reply:
This is to cover the case where the `Simulation` is initialized without any `unique_filters`. When you try to ask for the images, it ends up calling my method `_apply_sim_filters`, which itself asks for the `filter_indices`. So even if there are no filters, the `filter_indices` have to be set. 

The specific case where this happens is in `test_simulation.py`:
```python
class SingleSimTestCase(TestCase):
    """Test we can construct a length 1 Sim."""

    def setUp(self):
        self.sim = Simulation(
            n=1,
            L=8,
        )

    def testImage(self):
        """Test we can get an Image from a length 1 Sim."""
        _ = self.sim.images(0, 1)
```

Now that you've brought it up, though, I think we should also set `unique_filters = [IdentityFilter()]` when none is specified, as we do in, e.g. `ArraySource`. Would still need to set the `filter_indices` though:

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/3223d6c2c308af66f321ca2b0ddf3933045f904b/src/aspire/source/image.py#L789-L792

I'll make the change to be more in line with the above. Thanks!

Reply:
Ah, thanks for the clarification! The proposed change sounds good to me.